front-end/src/utils/deviceInformation.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Check if the device is a mobile or tablet 
+ */
+export const isMobileOrTablet = (): boolean => {
```

Comment:
Instead of calling this function every time, simply set this as a global boolean variable here. It never changes throughout the execution.